clang/include/clang/CConv/ConstraintResolver.h

```diff
@@ -48,8 +49,8 @@ class ConstraintResolver {
   bool containsValidCons(const CVarSet &CVs);
   bool isValidCons(ConstraintVariable *CV);
   // Try to get the bounds key from the constraint variable set.
-  bool resolveBoundsKey(CVarSet &CVs, BoundsKey &BK);
-  bool resolveBoundsKey(ConstraintVariable *CV, BoundsKey &BK);
+  bool resolveBoundsKey(const CVarSet &CVs, BoundsKey &BK);
+  bool resolveBoundsKey(CVarOption CV, BoundsKey &BK);
```

Comment:
Wondering about this one. Previously the method was a `ConstraintVariable *` which is itself a kind of "option." This new version is an optional `ConstraintVariable *`; why do we need another level of "optionality" ?